classes/phing/tasks/ext/PhpCSTask.php

```diff
@@ -103,9 +112,26 @@ public function setFile(PhingFile $file): void
 
     public function main()
     {
+        $project = $this->getProject();
+        /*
         if ($this->file === null) {
             throw new BuildException('Missing attribute "file".');
         }
+        */
+        if ($this->file === null && count($this->filesets) == 0) {
+            throw new BuildException('Missing both attribute "file" and "fileset".');
+        }
+        if ($this->file === null) {
+            // check filesets, and compile a list of files for phpcs to analyse
+            foreach ($this->filesets as $fileset) {
+                $scanner = $fileset->getDirectoryScanner($project);
+                $files = $scanner->getIncludedFiles();
+                $fromDir = $fileset->getDir($project);
+                foreach ($files as $file) {
```

Comment:
Is it possible to use the iterator here?

Reply:
I've switched over to using getIterator now, in changeset a4f34801cceb009da33e7869530571efa07d089c